src/components/character-controller.js

```diff
@@ -88,11 +86,30 @@ AFRAME.registerComponent("character-controller", {
     this.el.emit("snap_rotate_right");
   },
 
-  handleTeleport: function(event) {
-    const position = event.detail.newPosition;
-    const navPosition = event.detail.hitPoint;
-    this.resetPositionOnNavMesh(position, navPosition, this.el.object3D);
-  },
+  // We assume the rig is at the root, and its local position === its world position.
+  teleportTo: (function() {
```

Comment:
A quick not about what it means to teleport with regards to the rig and the players head might be helpful.